cmd/release-controller/sync.go

```diff
@@ -514,6 +517,101 @@ func (c *Controller) syncReady(release *Release) error {
 	return nil
 }
 
+// GitHubPR contains the necessary info to identify a GitHub PR (org, repo, and pr number)
+type GitHubPR struct {
+	Org  string
+	Repo string
+	PR   int
+}
+
+// BugzillaPR contains a bugzilla ID and the associated GitHub PR that resolves the bug
+type BugzillaPR struct {
+	bugzillaID int
+	githubPR   GitHubPR
+}
+
+var (
+	// GHRegex matches github PR links
+	GHRegex = regexp.MustCompile(`\(https://github.com/[[:alnum:]]+/[[:alnum:]|\-]+/pull/[\d]+\)`)
+	// BZRegex matches the markdown labels to bugzilla links
+	BZRegex = regexp.MustCompile(`\[Bug [\d]+\]`)
+	// BZAssignRegex matches the QA assignment comment made by the openshift-ci-robot
+	BZAssignRegex = regexp.MustCompile(`Requesting review from QA contact:[[:space:]]+/cc @[[:alnum:]]+`)
+)
+
+// getBugzillaPRs identifies bugzilla bugs and the associated github PRs fixed in a release from
+// a given markdown changelog generated by `oc adm release info --changelog=git-cache-path from-tag to-tag`
+func getBugzillaPRs(input string) []BugzillaPR {
+	lineRegex := regexp.MustCompile(fmt.Sprintf(`%s.+%s`, BZRegex, GHRegex))
+	numberRegex := regexp.MustCompile(`[\d]+`)
+	matches := lineRegex.FindAllString(input, -1)
+	var newArr []BugzillaPR
+	for _, match := range matches {
+		bz := BZRegex.FindString(match)
+		bzID := numberRegex.FindString(bz)
+		bzInt, err := strconv.Atoi(bzID)
+		if err != nil {
+			glog.Errorf("Failed to convert bugzilla ID %s to integer: %v", bzID, err)
+			continue
+		}
+		ghLink := GHRegex.FindString(match)
+		trimmedGHLink := strings.TrimPrefix(ghLink, "(https://github.com/")
+		trimmedGHLink = strings.TrimSuffix(trimmedGHLink, ")")
+		splitGHLink := strings.Split(trimmedGHLink, "/")
+		if len(splitGHLink) != 4 {
+			glog.Errorf("Could not properly split github URL: %s", trimmedGHLink)
+			continue
+		}
+		ghPRInt, err := strconv.Atoi(splitGHLink[3])
+		if err != nil {
+			glog.Errorf("Failed to convert github PR ID %s to integer: %v", splitGHLink[3], err)
+			continue
+		}
+		bzPR := BugzillaPR{
+			bugzillaID: bzInt,
+			githubPR: GitHubPR{
+				Org:  splitGHLink[0],
+				Repo: splitGHLink[1],
+				PR:   ghPRInt,
+			},
+		}
+		newArr = append(newArr, bzPR)
+	}
+	return newArr
+}
+
+// prApprovedByQA looks through PR comments and identifies if an assigned
+// QA contact lgtm'd the PR
+func prApprovedByQA(comments []github.IssueComment, reviews []github.Review) bool {
+	var lgtms, qaContacts []string
+	for _, comment := range comments {
+		if strings.Contains(comment.Body, "/lgtm") {
```

Comment:
Technically we should reuse the regex upstream to make sure we don't have false positives